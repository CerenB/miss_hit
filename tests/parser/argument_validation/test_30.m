% From https://www.mathworks.com/help/matlab/matlab_prog/function-argument-validation-1.html

function test_30(a, b)
    arguments
        a {mustBeA(a,'double'), mustBeDims(a,2)}
        b {mustBeSize(b,[5,3])}
    end

    % Function code
end

% Custom validator functions
function mustBeA(input,className)
    % Test for specific class
    if ~isa(input,className)
        error('Input must be of class double.');
    end
end

function mustBeSize(input,sizeDims)
    % Test for specific size
    if ~isequal(size(input),sizeDims)
        error(['Input must be of size ',num2str(sizeDims)]);
    end
end

function mustBeDims(input,numDims)
    % Test for number of dimensions
    if ~isequal(length(size(input)),numDims)
        error(['Input must have ',num2str(numDims),' dimensions.']);
    end
end
