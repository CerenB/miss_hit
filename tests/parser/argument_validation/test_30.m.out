----------------------------------------------------------------------
--  Parse tree for test_30.m
Root: Function_File
  Functions: Function_Definition
    Signature: Function_Signature
      Name: Identifier <test_30>
      Inputs: Identifier <a>
      Inputs: Identifier <b>
    Validation: Arguments Special_Block
      Items: Entity_Constraints
        Name: Identifier <a>
        Functions: Reference
          Name: Identifier <mustBeA>
          Arguments: Identifier <a>
          Arguments: Char_Array_Literal 'double'
        Functions: Reference
          Name: Identifier <mustBeDims>
          Arguments: Identifier <a>
          Arguments: Number_Literal <2>
      Items: Entity_Constraints
        Name: Identifier <b>
        Functions: Reference
          Name: Identifier <mustBeSize>
          Arguments: Identifier <b>
          Arguments: Matrix_Expression
            Rows: Row
              Items: Number_Literal <5>
              Items: Number_Literal <3>
    Body: Sequence_Of_Statements
  Functions: Function_Definition
    Signature: Function_Signature
      Name: Identifier <mustBeA>
      Inputs: Identifier <input>
      Inputs: Identifier <className>
    Body: Sequence_Of_Statements
      Statements: If_Statement
        Action: If Action
          Guard: Unary_Operation ~
            Expression: Reference
              Name: Identifier <isa>
              Arguments: Identifier <input>
              Arguments: Identifier <className>
          Body: Sequence_Of_Statements
            Statements: Naked_Expression_Statement
              Expression: Reference
                Name: Identifier <error>
                Arguments: Char_Array_Literal 'Input must be of class double.'
  Functions: Function_Definition
    Signature: Function_Signature
      Name: Identifier <mustBeSize>
      Inputs: Identifier <input>
      Inputs: Identifier <sizeDims>
    Body: Sequence_Of_Statements
      Statements: If_Statement
        Action: If Action
          Guard: Unary_Operation ~
            Expression: Reference
              Name: Identifier <isequal>
              Arguments: Reference
                Name: Identifier <size>
                Arguments: Identifier <input>
              Arguments: Identifier <sizeDims>
          Body: Sequence_Of_Statements
            Statements: Naked_Expression_Statement
              Expression: Reference
                Name: Identifier <error>
                Arguments: Matrix_Expression
                  Rows: Row
                    Items: Char_Array_Literal 'Input must be of size '
                    Items: Reference
                      Name: Identifier <num2str>
                      Arguments: Identifier <sizeDims>
  Functions: Function_Definition
    Signature: Function_Signature
      Name: Identifier <mustBeDims>
      Inputs: Identifier <input>
      Inputs: Identifier <numDims>
    Body: Sequence_Of_Statements
      Statements: If_Statement
        Action: If Action
          Guard: Unary_Operation ~
            Expression: Reference
              Name: Identifier <isequal>
              Arguments: Reference
                Name: Identifier <length>
                Arguments: Reference
                  Name: Identifier <size>
                  Arguments: Identifier <input>
              Arguments: Identifier <numDims>
          Body: Sequence_Of_Statements
            Statements: Naked_Expression_Statement
              Expression: Reference
                Name: Identifier <error>
                Arguments: Matrix_Expression
                  Rows: Row
                    Items: Char_Array_Literal 'Input must have '
                    Items: Reference
                      Name: Identifier <num2str>
                      Arguments: Identifier <numDims>
                    Items: Char_Array_Literal ' dimensions.'
----------------------------------------------------------------------
In test_30.m, line 3
| function test_30(a, b)
|                 ^ info: this token is not linked to the ast
In test_30.m, line 3
| function test_30(a, b)
|                   ^ info: this token is not linked to the ast
In test_30.m, line 3
| function test_30(a, b)
|                      ^ info: this token is not linked to the ast
In test_30.m, line 5
|         a {mustBeA(a,'double'), mustBeDims(a,2)}
|           ^ info: this token is not linked to the ast
In test_30.m, line 5
|         a {mustBeA(a,'double'), mustBeDims(a,2)}
|                               ^ info: this token is not linked to the ast
In test_30.m, line 5
|         a {mustBeA(a,'double'), mustBeDims(a,2)}
|                                                ^ info: this token is not linked to the ast
In test_30.m, line 6
|         b {mustBeSize(b,[5,3])}
|           ^ info: this token is not linked to the ast
In test_30.m, line 6
|         b {mustBeSize(b,[5,3])}
|                           ^ info: this token is not linked to the ast
In test_30.m, line 6
|         b {mustBeSize(b,[5,3])}
|                               ^ info: this token is not linked to the ast
In test_30.m, line 13
| function mustBeA(input,className)
|                 ^ info: this token is not linked to the ast
In test_30.m, line 13
| function mustBeA(input,className)
|                       ^ info: this token is not linked to the ast
In test_30.m, line 13
| function mustBeA(input,className)
|                                 ^ info: this token is not linked to the ast
In test_30.m, line 20
| function mustBeSize(input,sizeDims)
|                    ^ info: this token is not linked to the ast
In test_30.m, line 20
| function mustBeSize(input,sizeDims)
|                          ^ info: this token is not linked to the ast
In test_30.m, line 20
| function mustBeSize(input,sizeDims)
|                                   ^ info: this token is not linked to the ast
In test_30.m, line 23
|         error(['Input must be of size ',num2str(sizeDims)]);
|                                        ^ info: this token is not linked to the ast
In test_30.m, line 27
| function mustBeDims(input,numDims)
|                    ^ info: this token is not linked to the ast
In test_30.m, line 27
| function mustBeDims(input,numDims)
|                          ^ info: this token is not linked to the ast
In test_30.m, line 27
| function mustBeDims(input,numDims)
|                                  ^ info: this token is not linked to the ast
In test_30.m, line 30
|         error(['Input must have ',num2str(numDims),' dimensions.']);
|                                  ^ info: this token is not linked to the ast
In test_30.m, line 30
|         error(['Input must have ',num2str(numDims),' dimensions.']);
|                                                   ^ info: this token is not linked to the ast
MISS_HIT Summary: 1 file(s) analysed, everything seemes fine
